main.rb

```diff
@@ -0,0 +1,128 @@
+# ruby project.
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times { |i| yield(to_a[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each { |i| new_array << i if yield i }
+    new_array
+  end
+
+  def my_all?(args = nil)
+    if !args.nil? && !args.is_a?(Class)
+      if args.is_a?(Regexp)
+        my_each { |i| return false unless !i[args].nil? || i[args] == 1 }
+      else
+        my_each { |i| return false unless i == args }
+      end
+    elsif args.is_a?(Class)
+      my_each { |i| return false unless i.is_a?(args) }
+    elsif args.nil? && block_given?
+      my_each { |i| return false unless yield i }
+    else
+      my_each { |i| return false if i.nil? || i == false }
+    end
+    true
+  end
+
+  def my_any?(args = nil)
```

Comment:
my_any? method should return false if at least one of the collection is not true.
[Reference](https://ruby-doc.org/core-2.6.5/Enumerable.html).
![Screen Shot 2020-08-04 at 7 08 13 PM](https://user-images.githubusercontent.com/52430951/89357358-dde52d80-d685-11ea-82b0-a31555841ac3.png)
****
my_any? method when Regex is given as an argument it should returns true if any of the collection matches the Regex.
![Screen Shot 2020-08-04 at 7 12 34 PM](https://user-images.githubusercontent.com/52430951/89357563-79769e00-d686-11ea-9c7c-285b4cdedac4.png)

Reply:
##@davitomix thank you for the feedback. The requested change has been made.